app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -145,7 +145,7 @@ def handle_invalid_otp(type:, context: nil)
   def invalid_otp_error(type)
     case type
     when 'otp'
-      t('two_factor_authentication.invalid_otp')
+      t('two_factor_authentication.invalid_otp') + attempts_remaining_warning
```

Comment:
Minor: Rather than embed the space character into the translated text, and to be able to remove the `else` block altogether in `attempts_remaining_warning`, I wonder if we could treat these as distinct parts, and join them together only if the extra text is relevant.

```suggestion
      [t('two_factor_authentication.invalid_otp'), attempts_remaining_warning].select(:present?).join(' ')
```

```rb
  def attempts_remaining_warning
    return if attempts_count_remaining >= IdentityConfig.store.minimum_attempts_remaining_warning_count
    t(
      'two_factor_authentication.attempt_remaining_warning_html',
      count: attempts_count_remaining,
    )
  end
```